xdsl/dialects/arm/ops.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+from xdsl.ir import Operation
+from xdsl.irdl import IRDLOperation, irdl_op_definition, result_def
+
+from .register import IntRegisterType
+
+R1InvT = TypeVar("R1InvT", bound=IntRegisterType)
+
+
+class ARMOp(Operation, ABC):
```

Comment:
It might make sense to have the following?
```suggestion
class ARMOp(IRDLOperation, ABC):
```
Then you can remove `IRDLOperation` below.